instrumentation/opentelemetry-instrumentation-confluent-kafka/src/opentelemetry/instrumentation/confluent_kafka/__init__.py

```diff
@@ -327,26 +327,27 @@ def _uninstrument(self, **kwargs):
 
     @staticmethod
     def wrap_produce(func, instance, tracer, args, kwargs):
-        topic = kwargs.get("topic")
-        if not topic:
-            topic = args[0]
+        topic = KafkaPropertiesExtractor.extract_produce_topic(args, kwargs)
+        headers = KafkaPropertiesExtractor.extract_produce_headers(
+            args, kwargs
+        )
+        if headers is None:
+            headers = []
+            kwargs["headers"] = headers
+
+        partition = KafkaPropertiesExtractor.extract_produce_partition(
+            args, kwargs
+        )
 
-        span_name = _get_span_name("send", topic)
+        span_name = _get_span_name("publish", topic)
         with tracer.start_as_current_span(
             name=span_name, kind=trace.SpanKind.PRODUCER
         ) as span:
-            headers = KafkaPropertiesExtractor.extract_produce_headers(
-                args, kwargs
-            )
-            if headers is None:
-                headers = []
-                kwargs["headers"] = headers
-
-            topic = KafkaPropertiesExtractor.extract_produce_topic(args)
             _enrich_span(
                 span,
                 topic,
-                operation=MessagingOperationValues.RECEIVE,
+                partition,
+                operation="publish",
```

Comment:
Why is this a string instead of enum from semconv?
The RECEIVE in the old code is definitely incorrect though.